pymodbus/client/base.py

```diff
@@ -203,7 +205,12 @@ def callback_data(self, data: bytes, addr: tuple | None = None) -> int:
 
         returns number of bytes consumed
         """
-        self.framer.processIncomingPacket(data, self._handle_response, slave=0)
+        task = asyncio.create_task(
```

Comment:
This is client code and have nothing to do with the datastore which is purely server.

Reply:
Ahh I see that CI is now Green, one important step done.

You still need to limit your changes, especially so that NO client code is included.

But let me repeat, the idea of the PR is good and very welcome.

Reply:
I think I checked that sync client still works as expected, but I agree that change is quite wide.

I'll take another round to check more limited changes and try to reduce touching the client-used code changes.

Thanks for the quick comments.

Reply:
I am pretty sure, that adding 2 new async methods(), will meet your requirements, and limit all changes to the datastore directory + test.

Reply:
request me as reviewer when you are ready.

Reply:
I think the new round is closer to your proposal on approach and now it should be contained only on serverside and datastore.